app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java

```diff
@@ -685,77 +685,243 @@ private double getSumDateOffHaveSalary() {
                 + StringUtils.convertStringToDouble(mRequestOff.getWifeLaborLeave());
     }
 
-    private void setRequestDayOffTypesAttribute() {
-        mRequestDayOffTypesAttributes.clear();
-        if (mRequestOff.getAnnualLeave() != null
-                && !"".equals(mRequestOff.getAnnualLeave())
-                && !Constant.DEFAULT_DOUBLE_VALUE.equals(mRequestOff.getAnnualLeave())) {
-            mRequestOff.setNumberDayOffNormal(Double.parseDouble(mRequestOff.getAnnualLeave()));
+    private void setAttributeLeaveForMarriage() {
+        mRequestDayOffTypesAttribute = new OffRequest.RequestDayOffTypesAttribute();
+        mRequestDayOffTypesAttribute.setId(null);
+        mRequestDayOffTypesAttribute.setSpecialDayOffSettingId(TypeOfDays.LEAVE_FOR_MARRIAGE);
+        mRequestDayOffTypesAttribute.setNumberDayOff(mRequestOff.getLeaveForMarriage());
+        mRequestDayOffTypesAttribute.setDestroy(FALSE);
+        if (mRequestOff.getRequestDayOffTypes() != null
+                || mRequestOff.getRequestDayOffTypes().size() != 0) {
+            for (int i = 0; i < mRequestOff.getRequestDayOffTypes().size(); i++) {
+                if (mRequestOff.getRequestDayOffTypes().get(i).getSpecialDayOffSettingId()
+                        == Integer.parseInt(RequestOffViewModel.TypeOfDays.LEAVE_FOR_MARRIAGE)) {
+                    mRequestDayOffTypesAttribute.setId(
+                            mRequestOff.getRequestDayOffTypes().get(i).getId());
+                }
+            }
+            if (mRequestOff.getLeaveForMarriage() == null || "".equals(
+                    mRequestOff.getLeaveForMarriage()) || Constant.DEFAULT_DOUBLE_VALUE.equals(
+                    mRequestOff.getLeaveForMarriage())) {
+                mRequestDayOffTypesAttribute.setDestroy(TRUE);
+            }
         }
-        if (mRequestOff.getLeaveForMarriage() != null && !"".equals(
-                mRequestOff.getLeaveForMarriage())) {
-            mRequestDayOffTypesAttribute.setSpecialDayOffSettingId(TypeOfDays.LEAVE_FOR_MARRIAGE);
-            mRequestDayOffTypesAttribute.setNumberDayOff(mRequestOff.getLeaveForMarriage());
-            mRequestDayOffTypesAttribute.setDestroy(FALSE);
-            mRequestDayOffTypesAttributes.add(mRequestDayOffTypesAttribute);
+        mRequestDayOffTypesAttributes.add(mRequestDayOffTypesAttribute);
+    }
+
+    private void setAttributeLeaveForChildMarriage() {
+        mRequestDayOffTypesAttribute = new OffRequest.RequestDayOffTypesAttribute();
+        mRequestDayOffTypesAttribute.setId(null);
+        mRequestDayOffTypesAttribute.setSpecialDayOffSettingId(TypeOfDays.LEAVE_FOR_CHILD_MARRIAGE);
+        mRequestDayOffTypesAttribute.setNumberDayOff(mRequestOff.getLeaveForChildMarriage());
+        mRequestDayOffTypesAttribute.setDestroy(FALSE);
+        if (mRequestOff.getRequestDayOffTypes() != null
```

Comment:
ERROR: [BrokenNullCheck] Method call on object which may be null
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/basic.html#BrokenNullCheck